proparse/src/main/java/org/prorefactor/proparse/antlr4/PreproEval.java

```diff
@@ -0,0 +1,612 @@
+import org.prorefactor.proparse.antlr4.PreprocessorParser.DivideContext;
+import org.prorefactor.proparse.antlr4.PreprocessorParser.EqualsContext;
+import org.prorefactor.proparse.antlr4.PreprocessorParser.ExprInParenContext;
+import org.prorefactor.proparse.antlr4.PreprocessorParser.FalseExprContext;
+import org.prorefactor.proparse.antlr4.PreprocessorParser.GreaterEqualsContext;
+import org.prorefactor.proparse.antlr4.PreprocessorParser.GreaterThanContext;
+import org.prorefactor.proparse.antlr4.PreprocessorParser.Int64FunctionContext;
+import org.prorefactor.proparse.antlr4.PreprocessorParser.IntegerFunctionContext;
+import org.prorefactor.proparse.antlr4.PreprocessorParser.LesserEqualsContext;
+import org.prorefactor.proparse.antlr4.PreprocessorParser.LesserThanContext;
+import org.prorefactor.proparse.antlr4.PreprocessorParser.MatchesContext;
+import org.prorefactor.proparse.antlr4.PreprocessorParser.MinusContext;
+import org.prorefactor.proparse.antlr4.PreprocessorParser.ModuloContext;
+import org.prorefactor.proparse.antlr4.PreprocessorParser.NotContext;
+import org.prorefactor.proparse.antlr4.PreprocessorParser.NotEqualsContext;
+import org.prorefactor.proparse.antlr4.PreprocessorParser.NumberContext;
+import org.prorefactor.proparse.antlr4.PreprocessorParser.OrContext;
+import org.prorefactor.proparse.antlr4.PreprocessorParser.PlusContext;
+import org.prorefactor.proparse.antlr4.PreprocessorParser.PreproIfEvalContext;
+import org.prorefactor.proparse.antlr4.PreprocessorParser.ProversionFunctionContext;
+import org.prorefactor.proparse.antlr4.PreprocessorParser.QuotedStringContext;
+import org.prorefactor.proparse.antlr4.PreprocessorParser.RandomFunctionContext;
+import org.prorefactor.proparse.antlr4.PreprocessorParser.TrueExprContext;
+import org.prorefactor.proparse.antlr4.PreprocessorParser.UnaryMinusContext;
+import org.prorefactor.proparse.antlr4.PreprocessorParser.UnknownExprContext;
+import org.prorefactor.refactor.settings.IProparseSettings;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class PreproEval extends PreprocessorParserBaseVisitor<Object> {
+  private static final Logger LOGGER = LoggerFactory.getLogger(PreproEval.class);
+
+  /**
+   * Main entry point returning the evaluation of the preprocessor expression
+   * 
+   * @return A Boolean object
+   */
+  @Override
+  public Boolean visitPreproIfEval(PreproIfEvalContext ctx) {
+    LOGGER.trace("Entering visitPreproIfEval()");
+    Object o = visit(ctx.expr());
+    LOGGER.trace("Exiting visitPreproIfEval() with return value '{}'", o);
+    return (o != null) && getBool(o);
+  }
+
+  // ****
+  // Expr
+  // ****
+  @Override
+  public Object visitOr(OrContext ctx) {
+    boolean b1 = getBool(visit(ctx.expr(0)));
+    boolean b2 = getBool(visit(ctx.expr(1)));
+
+    return new Boolean(b1 || b2);
+  }
+
+  @Override
+  public Object visitAnd(AndContext ctx) {
+    return new Boolean(getBool(visit(ctx.expr(0))) && getBool(visit(ctx.expr(1))));
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove this "Boolean" constructor [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS2129)